src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -145,6 +183,34 @@ public bool Equals(ArraySegment<T> obj)
             return obj._array == _array && obj._offset == _offset && obj._count == _count;
         }
 
+        public ArraySegment<T> Slice(int index)
+        {
+            ThrowInvalidOperationIfDefault();
+            // Note: `index` is allowed to be negative so the start of this ArraySegment can be moved backwards.
```

Comment:
> I think we want to make sure indexing is blazing fast

Plain array is blazing fast. ArraySegment is not. It has extra overhead by design.

> It's also a pattern that doesn't exist in the BCL. I don't think we should introduce it here.

Agree.